### PR TITLE
net-vpn/i2p: add 2.6.0

### PR DESCRIPTION
**Pull request by tharvik**
bump to newer i2p. some rework to avoid java-ant-2 dep.

* drop java-ant-2
  * patch our classpath in a bunch of build.xml
  * add build dep on dev-java/ant
  * link a few system jar where possible
  * use `eant` directly
* avoid one useless sed while unbundling bcprov

keeping version 2.5.2-r1 until 2.6.0 is stablized

---

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

### CODE REVIEW

**Comment by vaukai** [comment]
> bump to newer i2p. some rework to avoid java-ant-2 dep.
> 
>     * drop java-ant-2
>       
>       * patch our classpath in a bunch of build.xml
>       * add build dep on dev-java/ant
>       * link a few system jar where possible
>       * use `eant` directly
> 
>     * avoid one useless sed while unbundling bcprov
> 
> 
> keeping version 2.5.2-r1 until 2.6.0 is stablized

thanks for jumping on this train. my [todo list](https://qa-reports.gentoo.org/output/eapi-per-eclass/java-ant-2.eclass/) is getting shorter ...
`bt747` and `c3p0` still to be converted, [javacup trouble](https://bugs.gentoo.org/showdependencytree.cgi?id=936274) to be solved.
@fordfrog with some luck we'd get completely rid on `java-ant-2.eclass`?

File: net-vpn/i2p/i2p-2.6.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+		-path "./apps/jetty/jetty-distribution-*" # need to package jetty
+	( cat >> override.properties || die 'set unbundled properties' ) <<- EOF
+		require.gettext=true
+		with-libgetopt-java=true
+		with-libjakarta-taglibs-standard-java=true
+		with-libjson-simple-java=true
+		with-libtomcat9-java=true
+		with-gettext-base=true
+		# with-geoip-database=true need std geoip use
+		# with-libjetty9-java=true needs a jetty ebuild
```
Comment (vaukai) [inline net-vpn/i2p/i2p-2.6.0.ebuild:98]: doesn't it also need [ant.build.javac.source and ant.build.javac.target](https://github.com/gentoo/gentoo/pull/37945/files#diff-b94ac2e4fdd4bb3cb637d47df5404de960aa7e05dd54e998a5c42376f6ef707fR62-R63)?
to get this working, any eventually hardcoded `-source` and `-target` attributes must be removed from the `javac` elements in `build.xml`.
this was another part of `BSFIX` provided by `java-ant-2`.

Comment (tharvik) [reply net-vpn/i2p/i2p-2.6.0.ebuild:98]: good catch, it was hardcoded everywhere. added a fix in latest commit.

**Comment by tharvik** [comment]
thanks for the review.

* added a find/sed to remove target & source everywhere
* remove some uneeded direct deps (are included transitively and not direclty used in code)
* `$PV` prefix patch

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**